src/azul/service/responseobjects/buffer.py

```diff
@@ -0,0 +1,46 @@
+from logging import getLogger
+from io import StringIO
+from threading import Lock
+from typing import Callable
+
+logger = getLogger(__name__)
+
+
+class Buffer(StringIO):
+
+    def __init__(self, limit_size: int, inbetween_callback: Callable):
+        super(Buffer, self).__init__()
+        self.limit_size = limit_size
+        self.inbetween_callback = inbetween_callback
+        self.__remaining_size = 0
+        self.__total_size = 0
+        self.__pointer_lock = Lock()
+
+    def write(self, s: str):
+        self.__pointer_lock.acquire()
+        byte_count = len(s.encode())
```

Comment:
Decide if Buffer deals with bytes or with strings. You shouldn't have to convert back and forth. For bytes there is also BytesIO.

Reply:
Good point. It probably won't. Consider wrapping the instance of this class in an `io.TextIOWrapper(write_through=True, encoding="utf8")` so this class doesn't have to worry about encoding.